px4_ros2_cpp/src/control/setpoint_types/experimental/trajectory.cpp

```diff
@@ -90,8 +91,16 @@ SetpointBase::Configuration TrajectorySetpointType::getConfiguration()
   config.rates_enabled = true;
   config.attitude_enabled = true;
   config.acceleration_enabled = true;
-  config.velocity_enabled = true;
-  config.position_enabled = true;
+
+  if(!_local_position_is_optional){
+    config.position_enabled = true;
+    config.velocity_enabled = true;
+  }
+  else {
+    config.position_enabled = false;
+    config.velocity_enabled = false;
+  }
```

Comment:
Did you test this? This means the velocity controller is disabled.
You'll need to add a flag `local_position_is_optional` to the [Configuration struct](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/px4_ros2_cpp/include/px4_ros2/common/setpoint_base.hpp#L45). And then use it here: https://github.com/Auterion/px4-ros2-interface-lib/blob/main/px4_ros2_cpp/src/components/mode.cpp#L247